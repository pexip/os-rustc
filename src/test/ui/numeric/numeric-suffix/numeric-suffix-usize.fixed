// run-rustfix

fn foo<N>(_x: N) {}

fn main() {
    foo::<usize>(42_usize);
    foo::<usize>(42_usize);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<usize>(42_usize);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<usize>(42_usize);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<usize>(42_usize);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<usize>(42_usize);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<usize>(42_usize);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<usize>(42_usize);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<usize>(42_usize);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<usize>(42_usize);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<usize>(42usize);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<usize>(42usize);
    //~^ ERROR mismatched types
    //~| NOTE expected
}
