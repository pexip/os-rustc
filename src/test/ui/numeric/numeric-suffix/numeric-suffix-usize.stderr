error[E0308]: mismatched types
  --> $DIR/numeric-suffix-usize.rs:7:18
   |
LL |     foo::<usize>(42_u64);
   |                  ^^^^^^ expected `usize`, found `u64`
   |
help: change the type of the numeric literal from `u64` to `usize`
   |
LL |     foo::<usize>(42_usize);
   |                     ~~~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-usize.rs:10:18
   |
LL |     foo::<usize>(42_u32);
   |                  ^^^^^^ expected `usize`, found `u32`
   |
help: change the type of the numeric literal from `u32` to `usize`
   |
LL |     foo::<usize>(42_usize);
   |                     ~~~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-usize.rs:13:18
   |
LL |     foo::<usize>(42_u16);
   |                  ^^^^^^ expected `usize`, found `u16`
   |
help: change the type of the numeric literal from `u16` to `usize`
   |
LL |     foo::<usize>(42_usize);
   |                     ~~~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-usize.rs:16:18
   |
LL |     foo::<usize>(42_u8);
   |                  ^^^^^ expected `usize`, found `u8`
   |
help: change the type of the numeric literal from `u8` to `usize`
   |
LL |     foo::<usize>(42_usize);
   |                     ~~~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-usize.rs:19:18
   |
LL |     foo::<usize>(42_isize);
   |                  ^^^^^^^^ expected `usize`, found `isize`
   |
help: change the type of the numeric literal from `isize` to `usize`
   |
LL |     foo::<usize>(42_usize);
   |                     ~~~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-usize.rs:22:18
   |
LL |     foo::<usize>(42_i64);
   |                  ^^^^^^ expected `usize`, found `i64`
   |
help: change the type of the numeric literal from `i64` to `usize`
   |
LL |     foo::<usize>(42_usize);
   |                     ~~~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-usize.rs:25:18
   |
LL |     foo::<usize>(42_i32);
   |                  ^^^^^^ expected `usize`, found `i32`
   |
help: change the type of the numeric literal from `i32` to `usize`
   |
LL |     foo::<usize>(42_usize);
   |                     ~~~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-usize.rs:28:18
   |
LL |     foo::<usize>(42_i16);
   |                  ^^^^^^ expected `usize`, found `i16`
   |
help: change the type of the numeric literal from `i16` to `usize`
   |
LL |     foo::<usize>(42_usize);
   |                     ~~~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-usize.rs:31:18
   |
LL |     foo::<usize>(42_i8);
   |                  ^^^^^ expected `usize`, found `i8`
   |
help: change the type of the numeric literal from `i8` to `usize`
   |
LL |     foo::<usize>(42_usize);
   |                     ~~~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-usize.rs:34:18
   |
LL |     foo::<usize>(42.0_f64);
   |                  ^^^^^^^^ expected `usize`, found `f64`
   |
help: change the type of the numeric literal from `f64` to `usize`
   |
LL |     foo::<usize>(42usize);
   |                    ~~~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-usize.rs:37:18
   |
LL |     foo::<usize>(42.0_f32);
   |                  ^^^^^^^^ expected `usize`, found `f32`
   |
help: change the type of the numeric literal from `f32` to `usize`
   |
LL |     foo::<usize>(42usize);
   |                    ~~~~~

error: aborting due to 11 previous errors

For more information about this error, try `rustc --explain E0308`.
